### DISCUSSION TITLE
How to start a new line in a multi-lined text box or label?

### DISCUSSION THREAD

**Discussion by deangoddard** [tznind/TerminalGuiDesigner#303] [Q&A]
I wanted to try adding some ASCII art to my main or start view, only to discover that adding text to my label pressing <Enter> doesn't create a new line instead it jumps out of editing mode.  Even tried Shit+Enter or Alt+Enter but yeah nah...

Just curious if I was doing something wrong?

**Comment by deangoddard** [comment]
Just playing around and found out the multi-lined textboxes are supported.

**Reply by deangoddard** [reply]
Will raise this as a suggestion.

**Comment by tznind** [comment]
Interesting, I don't think I have considered this use case.

Shift enter for `\n` makes sense. I think it would need to go in the 'Edit UI' for the `Text` property.

It could probably be implemented in TerminalGuiDesigner but might make more sense in main library when a TextView has AllowEnter false.

What do you think @BDisp ?

@deangoddard are you using v1 or v2?

In the mean time i think only workaround would be set property in `MyView.cs` constructor or manually modify `MyView.Designer.cs` (less than ideal)

**Reply by deangoddard** [reply]
v2.

Seen Terminal GUI around for a while now, and with the designer thought I might give v2 a try on my CLI utility.   I need a quick UI and WinForms is too old, WPF not future proof, WinUI pretty but heavy going for what I need.  So newb to TUI, but worked on green screen tech when I was 20 something...  Nothing beats it for dev and performance speed.
